src/interactive-window/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -15,7 +15,11 @@ import { getInteractiveCellMetadata } from '../../../interactive-window/helpers'
 import { KernelDebugAdapterBase } from '../../../kernels/debugger/kernelDebugAdapterBase';
 import { InteractiveCellMetadata } from '../../editor-integration/types';
 import { IDebugService } from '../../../platform/common/application/types';
-
+/**
+ * KernelDebugAdapter listens to debug messages in order to translate file requests into real files
+ * (Interactive Window generally executes against a real file)
+ * Files from the kernel are pointing to cells, whereas the user is looking at a real file.
```

Comment:
Does it mean that the difference between notebook debugging and interactive window debugging is IW has one more step: mapping DAP file requests from cell uris to file uri (python file)?

Reply:
Pretty much.

Reply:
At least once we switch to using the IPykernel6 debugger. The old debugger is slightly different as it does attach and detach at key spots and loads debugpy itself.